ccdproc/shutter_correction.py

```diff
@@ -0,0 +1,354 @@
+from __future__ import (print_function, division, absolute_import,
+                        unicode_literals)
+
+import os
+
+import numpy as np
+import astropy.units as u
+from astropy.nddata import StdDevUncertainty
+from astropy.modeling import models, fitting
+
+from .core import CCDData, subtract_bias
+from .ccddata import fits_ccddata_reader, fits_ccddata_writer
+from .combiner import combine
+from .image_collection import ImageFileCollection
+
+
+def GaladiEnriqez1995(combined_flats, exptimekey='EXPTIME', output=None):
+    '''
+    Determine the shutter correction map given a set of bias subtracted flats
+    taken under constant illumination.
+    
+    Parameters
+    ----------
+    combined_flats : `list` containing `~ccdproc.CCDData` objects
+        Set of bias subtracted flats of different exposure times taken under
+        constant illumination.
+    
+    exptimekey : `string`, optional, defaults to 'EXPTIME'
+        The header keyword which contains the exposure time of the image in
+        seconds.
+
+    output : `str`, optional
+        The filename to write the output shutter map to.  This will be a FITS
+        file with two extensions.  The first contains the shutter map values
+        (in seconds) and the second contains the uncertainty in those values.
+
+    Notes
+    -----
+    This code follows the implementation described in Galadi-Enriquez, D.,
+    Jordi, C., & Trullols, E. (1995). "Effects of Shutter Timing on CCD
+    Photometry". IAU Symposium No. 167, 167, 327.
+
+    Returns
+    -------
+    shutter_map : `~ccdproc.CCDData`
+        The shutter correction map in units of seconds.
+    '''
+    n = len(combined_flats)
+    exptimes = np.ma.MaskedArray([f.header[exptimekey] for f in combined_flats],
```

Comment:
I wouldn't bother creating a `MaskedArray` here if you manually set the mask.

```
try:
    exptimes = [f.meta[exptimekey] for f in combined_flats]
except KeyError:
    raise ValueError('exposure time keyword {} not found in at least one input'.format(exptimekey))
except AttributeError:
    raise TypeError('at least one input is no `CCDData` object.')
exptimes = np.array(exptimes)
idx_exp_max = np.argmax(exptimes)
exp_max = np.max(exptimes)
exp_sum = np.sum(exptimes) - exp_max  # the sum without the maximum.
```

You don't have to do this, I generally find it helpful to have more expressive variables around (if they are used later).